app/graphql/types/system.rb

```diff
@@ -54,6 +54,18 @@ def rules_failed(args = {})
       end
     end
 
+    def test_result_profiles
+      ::CollectionLoader.for(object.class, :test_result_profiles)
+                        .load(object)
+    end
+
+    def policies
+      ::CollectionLoader.for(object.class, :policies)
+                        .load(object).then do |policies|
+        policies.map(&:initial_profile)
```

Comment:
You can continue with `CollectionLoader` here, basically chaining the promises. Unfortunately it would likely required a dual implementation of `initial_profile`, however it would speed up the overall GraphQL querying.